### PR TITLE
Avoid propagating revert in cron upkeep

### PR DESCRIPTION
**Pull request by RyanRHall**
The changes in this PR help address customer confusion around the functionality of the cron upkeep. We previously designed the cron upkeep to _avoid_ sending failing transactions, in an effort to help users save $$. This would create confusion for customers though, because they would expect to see a tx at a given time, but then not see it. Customers didn't realize that keepers were trying to help them, and we didn't provide any other mechanism to alert them that their contract was reverting.

Instead, we would like to restructure the cron upkeep to *fail fast and loud*. And nothing is louder than burning money :). Instead of propagating customer contract reverts, we will now rescue them. Customers should be able to see txs fire exactly on the cron routine they specify, and they can use the failing txs to help debug their contracts.

### CODE REVIEW

File: contracts/src/v0.8/upkeeps/CronUpkeep.sol [hunk 1]
```diff
@@ -37,12 +36,11 @@ import {getRevertMsg} from "../utils/utils.sol";
 contract CronUpkeep is KeeperCompatibleInterface, KeeperBase, ConfirmedOwner, Pausable, Proxy {
   using EnumerableSet for EnumerableSet.UintSet;
 
-  event CronJobExecuted(uint256 indexed id, uint256 timestamp);
+  event CronJobExecuted(uint256 indexed id, bool success);
```
Comment (RyanRHall) [inline contracts/src/v0.8/upkeeps/CronUpkeep.sol:39]: timestamp is a useless field - you can get it from the block the tx was included in if needed

File: contracts/src/v0.8/upkeeps/CronUpkeep.sol [hunk 2]
```diff
@@ -92,11 +90,8 @@ contract CronUpkeep is KeeperCompatibleInterface, KeeperBase, ConfirmedOwner, Pa
     );
     validate(id, tickTime, target, handler);
     s_lastRuns[id] = block.timestamp;
-    (bool success, bytes memory payload) = target.call(handler);
-    if (!success) {
-      revert CallFailed(id, getRevertMsg(payload));
-    }
-    emit CronJobExecuted(id, block.timestamp);
+    (bool success, ) = target.call(handler);
+    emit CronJobExecuted(id, success);
```
Comment (RyanRHall) [inline contracts/src/v0.8/upkeeps/CronUpkeep.sol:94]: we _could_ include revert data in the event, but I'm unsure of the value add

File: contracts/test/v0.8/CronUpkeep.test.ts [hunk 3]
```diff
@@ -99,7 +98,7 @@ describe('CronUpkeep', () => {
     ) // the typechain factory that creates the cron factory contract
     cronFactoryContract = await cronFactoryContractFactory.deploy()
     const fs = cronReceiver1.interface.functions
-    handler1Sig = utils.id(fs['handler1()'].format('sighash')).slice(0, 10) // TODO this seems like an ethers bug
+    handler1Sig = utils.id(fs['handler1()'].format('sighash')).slice(0, 10)
```
Comment (RyanRHall) [inline contracts/test/v0.8/CronUpkeep.test.ts:101]: not a bug lol

File: contracts/src/v0.8/upkeeps/CronUpkeep.sol (hunk 2 above)
Comment (wentzeld) [reply contracts/src/v0.8/upkeeps/CronUpkeep.sol:94]: @RyanRHall you added a revert reason on registry right?

File: contracts/src/v0.8/upkeeps/CronUpkeep.sol [hunk 4]
```diff
@@ -92,11 +90,8 @@ contract CronUpkeep is KeeperCompatibleInterface, KeeperBase, ConfirmedOwner, Pa
     );
     validate(id, tickTime, target, handler);
```
Comment (wentzeld) [inline contracts/src/v0.8/upkeeps/CronUpkeep.sol:91]: @RyanRHall can we remove this ?? this validate adds a lot of extra gas.

File: contracts/src/v0.8/upkeeps/CronUpkeep.sol (hunk 2 above)
Comment (infiloop2) [reply contracts/src/v0.8/upkeeps/CronUpkeep.sol:94]: I would say to not have it unless we see a need. It adds extra gas and code size

Comment (infiloop2) [reply contracts/src/v0.8/upkeeps/CronUpkeep.sol:94]: also our previous method `getRevertMsg` only works `require` errors and not really for custom error signatures

**Review by infiloop2 — DISMISSED** [review]
looks good!

**Review by se3000 — APPROVED** [review]

**Pull request merged**